controllers/v1/solutions.js

```diff
@@ -16,64 +16,1066 @@ module.exports = class  Solutions extends Abstract{
+            "5fd1b52ab53a6416aaeefc80",
+            "5fd098e2e049735a86b748ac"
+        ],
+        "roles": "BEO"
+    }
+}
+
+   * @returns {JSON}
+   */
+
+   async update(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let solutionData = await solutionsHelper.update(
+          req.params._id, 
+          req.body, 
+          req.userDetails.userInformation.userId,
+          true                                              //this is true for when its called via API calls
+        );
+
+        return resolve(solutionData);
+      }
+      catch (error) {
+        reject({
+          status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+          message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+          errorObject: error
+        })
+      }
+    })
+  } 
+
+    /**
+    * @api {post} /project/v1/solutions/list?page=:page&limit=:limit&search=:search
     * @apiVersion 1.0.0
-    * @apiName Create solution
+    * @apiName Add roles in solutions
     * @apiGroup Solutions
     * @apiParamExample {json} Request-Body:
+    {
+      "isReusable": false
+    }
+
+    * @apiHeader {String} X-authenticated-user-token Authenticity token
+    * @apiSampleRequest /project/v1/solutions/list?page=1&limit=1&search=&type=improvementProject
+    * @apiUse successBody
+    * @apiUse errorBody
+    * @apiParamExample {json} Response:
+    * {
+        "message": "Solutions lists fetched successfully",
+        "status": 200,
+        "result": {
+            "data": [
+                {
+                    "_id": "5d15b0d7463d3a6961f91748",
+                    "externalId": "LSAS-Dream A Dream-2019-TEMPLATE",
+                    "name": "Life Skills Assessment Survey",
+                    "description": "Life Skills Assessment Survey"
+                }
+            ],
+            "count": 1
+        }
+      }
+
+    */
+
+   /**
+   * List solutions.
+   * @method
+   * @name list
+   * @param {Object} req - requested data.
+   * @param {String} req.query.type - solution type.
+   * @returns {JSON}
+   */
+
+   async list(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        let solutionData = await solutionsHelper.list(
+          req.query.type,                   //mandatory field to be given 
+          req.query.subType ? req.query.subType : "",
+          req.body,
+          req.pageNo,
+          req.pageSize,
+          req.searchText
+        );
+
+        return resolve(solutionData);
+      }
+      catch (error) {
+        reject({
+          status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+          message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+          errorObject: error
+        })
+      }
+    })
+  }
+
+
+    /**
+    * @api {post} /project/v1/solutions/addRolesInScope/:solutionId Add roles in solutions
+    * @apiVersion 1.0.0
+    * @apiName Add roles in solutions
```

Comment:
@Prajwal17Tunerlabs change it to add roles in scope

Reply:
changed the title